servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -186,13 +191,8 @@ private void handleMasterCommand(Command cmd) throws Exception {
       // Master requests blocks to be removed from Tachyon managed space.
       case Free:
         for (long block : cmd.mData) {
-          try {
-            mBlockDataManager.removeBlock(Users.MASTER_COMMAND_USER_ID, block);
-          } catch (NotFoundException nfe) {
-            // TODO: Throw a better exception here
-            // NotFoundException will be thrown if the block is unable to be locked.
-            LOG.warn("Failed master free block cmd for: " + block + " due to concurrent read.");
-          }
+          mFixedExecutionService.execute(new BlockRemover(mBlockDataManager,
```

Comment:
@zhljen @calvinjia One more question, is that really good enough to create a single thread for each block(here is up to `DEFAULT_BLOCK_REMOVER_POOL_SIZE` concurrent block remover thread) remove operation in the for-loop? It may bring considerable resource consumption. How about just pass the `cmd.mData` list to the one `block-remove-thread` ? The remove-thread will also execute asynchronous without blocking the `handleMasterCommand` method but with less resources cost. Thanks!